theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -63,3 +108,310 @@ def repoclosures(repo, releases, version) {
+            }
+
+            steps.executes["${os}-${project}"] = {
+                execute_pbuilder(execute_pbuilder_dir, os, version)
+            }
+
+            if (type == 'plugin' && !pull_request) {
+                suite = 'plugins'
+                component = version =~ 'develop' ? 'nightly' : version
+                release_type = 'release'
+            } else if (type == 'plugin' && pull_request) {
+                suite = 'plugins'
+                component = repoowner
+                release_type = 'stage'
+                echo "scratch build: uploading to stagingdeb/${suite}/${component}"
+            } else {
+                suite = os
+                component = "${repoowner}-${version}"
+                release_type = 'stage'
+                echo "scratch build: uploading to stagingdeb/${suite}/${component}"
+            }
+
+            if (!repos.containsKey("${suite}-${component}")) {
+                repos["${suite}-${component}"] = [suite: suite, component: component, deb_paths: [], type: release_type]
+            }
+            repos["${suite}-${component}"].deb_paths.add("${deb_path}/*deb}")
+        }
+    }
+
+    repos.each { key, value ->
+        if (value.type == 'stage') {
+            steps.rsync_packages[key] = {
+                rsync_to_debian_stage(value.suite, value.component, value.deb_paths)
+            }
+        } else {
+            steps.rsync_packages[key] = {
+                rsync_to_debian_release(value.suite, value.component, value.deb_paths)
+            }
+        }
+    }
+
+    return steps
+}
+
+def deb_build_dir(type, project, os = null) {
+    def build_dir
+
+    if (type == 'core') {
+        build_dir = "debian/${os}/build-${project}"
+    } else if (type == 'dependency') {
+        build_dir = "dependencies/${os}/build-${project}"
+    } else if (type == 'plugin') {
+        build_dir = "plugins/build-${project}"
+    } else {
+        echo "Unsupported type specified: ${type}"
+        exit 1
+    }
+
+    return build_dir
+}
+
+def setup_sources_deb(type, project, os, version, repoowner, pull_request) {
+
+    if (type == 'core') {
+        setup_sources_core(project, os, version, repoowner, pull_request)
+    } else if (type == 'dependency') {
+        setup_sources_dependency(project, os, version, repoowner, pull_request)
+    } else if (type == 'plugin') {
+        setup_sources_plugin(project, os, version, repoowner, pull_request)
+    } else {
+        echo "Unsupported type specified: ${type}"
+        exit 1
+    }
+
+}
+
+def setup_sources_core(os, project, version, repoowner, pull_request = false) {
+    def package_version = debian_package_version("debian/${os}/${project}/changelog")
+    def build_dir = deb_build_dir('core', project, os)
+
+    package_version = package_version.replaceAll('~rc', '-rc')
+    last_commit = package_version
+
+    dir(build_dir) {
+        if (version =~ 'develop') {
```

Comment:
why `=~` and not `=`? also this should be `nightly`, not `develop`, for the same reasons as in https://github.com/theforeman/jenkins-jobs/pull/96/files#r700835168

Reply:
Holdovers from trying to translate shell scripts to groovy. Thanks for the catch.